pttbbs/file.go

```diff
@@ -113,6 +130,76 @@ func OpenFileHeaderFile(filename string) ([]*FileHeader, error) {
 
 }
 
+func AppendFileHeaderFileRecord(filename string, newFileHeader *FileHeader) error {
+
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	err = filelock.Lock(f)
+	if err != nil {
+		// File is locked
+		return err
+	}
+
+	dirs := strings.Split(filename, "/")
+	newFilename := filename[:len(filename)-len(dirs[len(dirs)-1])] + newFileHeader.Filename()
```

Comment:
windows 會 fail 應該是因為這邊出錯

panic: runtime error: index out of range [2] with length 2 [recovered]
	panic: runtime error: index out of range [2] with length 2

Reply:
目前這段已拔除，但 windows 仍然會 fail
另外想問一下要如何看到比較詳細的錯誤資訊，像是下圖
```
panic: runtime error: index out of range [2] with length 2 [recovered]
panic: runtime error: index out of range [2] with length 2
```

Reply:
我實際用 windows 測試的結果是
```
The process cannot access the file because it is being used by another process.
```
看來是之前寫的 OpenFileHeaderFile 忘了 close file，更正後看來是沒問題了～